### PR TITLE
Maven `RemoveExclusion` recipe should preserve comments

### PR DESCRIPTION
**Pull request by knutwannheden**
Change `RemoveExclusion` recipe to use `Tag#getContent()` rather than `Tag#getChildren()` so that comments are preserved.

### CODE REVIEW

**Comment by knutwannheden** [comment]
I created this PR based on finding in the context of #2386.

**Review by pway99 — COMMENTED** [review]
This is great, thanks Knut. 
One suggestion: the recipe should finally check and delete the exclusions tag if its only contents are comments.

File: rewrite-maven/src/main/java/org/openrewrite/maven/RemoveExclusion.java [hunk 1]
```diff
@@ -66,17 +68,19 @@ public Xml visitTag(Xml.Tag tag, ExecutionContext executionContext) {
                 if (isDependencyTag(groupId, artifactId) || isManagedDependencyTag(groupId, artifactId)) {
                     Optional<Xml.Tag> maybeExclusions = tag.getChild("exclusions");
                     if (maybeExclusions.isPresent()) {
-                        return tag.withContent(ListUtils.map(tag.getChildren(), child -> {
-                            if("exclusions".equals(child.getName())) {
-                                Xml.Tag e = child;
+                        return tag.withContent(ListUtils.map((List<Content>) tag.getContent(), child -> {
```
Comment (pway99) [inline rewrite-maven/src/main/java/org/openrewrite/maven/RemoveExclusion.java:71]: just a little comment to keep the ide happy
```suggestion
                        //noinspection unchecked
                        return tag.withContent(ListUtils.map((List<Content>) tag.getContent(), child -> {
```

**Comment by knutwannheden** [comment]
> This is great, thanks Knut. One suggestion: the recipe should finally check and delete the `exclusions` tag if its only contents are comments.
> 
> We are about to start the release process, so I will merge the changes. Thanks again

You are right about your comment. Should I create a follow-up PR for that?

**Comment by pway99** [comment]
No, that's ok I went ahead and followed it up with a few additions :)

**Pull request merged**